### ISSUE TITLE
NPE if recording no longer has original channel information (Edge Case)

### ISSUE THREAD

**Issue by billmeek** [MythTV-Clients/MythTV-Android-Frontend#245]
<pre>
12-03 15:08:15.863: E/RecordedHelperV28(5580): process : non UTF-8 exception 
12-03 15:08:15.863: E/RecordedHelperV28(5580): java.lang.NullPointerException
12-03 15:08:15.863: E/RecordedHelperV28(5580):  at org.mythtv.service.channel.v28.ChannelHelperV28.convertChannelInfoToContentValues(ChannelHelperV28.java:495)
12-03 15:08:15.863: E/RecordedHelperV28(5580):  at org.mythtv.service.channel.v28.ChannelHelperV28.processChannel(ChannelHelperV28.java:315)
12-03 15:08:15.863: E/RecordedHelperV28(5580):  at org.mythtv.service.dvr.v28.RecordedHelperV28.load(RecordedHelperV28.java:294)
12-03 15:08:15.863: E/RecordedHelperV28(5580):  at org.mythtv.service.dvr.v28.RecordedHelperV28.downloadRecorded(RecordedHelperV28.java:220)
12-03 15:08:15.863: E/RecordedHelperV28(5580):  at org.mythtv.service.dvr.v28.RecordedHelperV28.process(RecordedHelperV28.java:126)
12-03 15:08:15.863: E/RecordedHelperV28(5580):  at org.mythtv.service.dvr.RecordedService.onHandleIntent(RecordedService.java:114)
12-03 15:08:15.863: E/RecordedHelperV28(5580):  at android.app.IntentService$ServiceHandler.handleMessage(IntentService.java:65)
12-03 15:08:15.863: E/RecordedHelperV28(5580):  at android.os.Handler.dispatchMessage(Handler.java:99)
12-03 15:08:15.863: E/RecordedHelperV28(5580):  at android.os.Looper.loop(Looper.java:137)
12-03 15:08:15.863: E/RecordedHelperV28(5580):  at android.os.HandlerThread.run(HandlerThread.java:60)
</pre>

To duplicate

<pre>
Make a 'normal' recording
Remove the original channel and source
e.g. if a backend has been reconfigured with different tuners
</pre>

``` sql
SELECT * FROM channel WHERE chanid = 3625;
```

Returns the proper information about the recording.

The data in the recorded MAF table appears to be correct.

The following snippet is returned from: 

```
http://mc0:6544/Dvr/GetRecordedList?TitleRegEx=Failing ProgramName
```

Note the empty ChanNum.

``` xml
<Channel>
<ChanId>3625</ChanId>
<ChanNum/>
<CallSign/>
<IconURL/>
<ChannelName/>
<MplexId>0</MplexId>
<TransportId>0</TransportId>
<ServiceId>0</ServiceId>
<NetworkId>0</NetworkId>
<ATSCMajorChan>0</ATSCMajorChan>
<ATSCMinorChan>0</ATSCMinorChan>
```

In MAF 0.28, the exception fires here:

``` java
    public ContentValues convertChannelInfoToContentValues( final LocationProfile locationProfile, final ChannelInfo channelInfo ) {
//      Log.v( TAG, "convertChannelToContentValues : enter" );

        String formattedChannelNumber = formatChannelNumber( channelInfo.getChanNum() );
        if( formattedChannelNumber.startsWith( "." ) ) { <------------------------- NPE
            formattedChannelNumber = formattedChannelNumber.substring( 1 );
        }
```

**Comment by billmeek** [comment]
Corrected typos in the original post. Also, there is no entry in the MAF channel
table for the channel above.

``` sql
SELECT * FROM channel WHERE chan_id  > 3000 and chan_id < 4000;
```

returns nothing.

**Comment by billmeek** [comment]
Fixed in 8e029ac and ae942ac. Also showed up when an 'old' recording
had a chanid that mapped to an invalid channum e.g. _50-1.
